app/controllers/admin/members_controller.rb

```diff
@@ -1,12 +1,44 @@
+require "csv"
+
 module Admin
   class MembersController < ::MembersController
-    http_basic_authenticate_with MembershipRegister.basic_auth_options
-
-    respond_to :csv, only: :index
+    http_basic_authenticate_with(
+      name: ENV.fetch("AUTH_NAME") { "secretary" },
+      password: ENV.fetch("AUTH_PASSWORD") {
+        password = SecureRandom.hex
+        Rails.logger.info("You have not set a password for the admin area, your temporary password is: #{password}")
```

Comment:
Should this be in `config/development.rb` instead? We wouldn't want to accidentally log the password in plain-text on production.

```ruby
# config/development.rb
config.admin_password =  ENV.fetch("AUTH_PASSWORD") {
         password = SecureRandom.hex
         Rails.logger.info("You have not set a password for the admin area, your temporary password is: #{password}")
}

# config/production.rb – the app should fail to boot if this is missing
config.admin_password = ENV.fetch("AUTH_PASSWORD")

# app/controllers/admin/members_controller.rb
http_basic_authenticate_with(password: Rails.configuration.admin_password)
```

Reply:
I prefer to avoid environment specific ENV vars.

I changed the implementation, if there is no ENV var:

* it just notifies there is no ENV var
* sets the password to a random one, but does not show it


This way the production environment will be protected anyway in case we forget to configure.